src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -305,4 +350,19 @@ private void constructSearchQueryRecord(final SearchPhaseContext context, final
         }
     }
 
+    /**
+     * Validate the index name for excluded indices
+     * @param excludedIndices list of index to validate
+     */
+    public void validateExcludedIndices(List<String> excludedIndices) {
```

Comment:
Check that excludedIndices is not null or add `@NonNull` tag

Reply:
Let's remove `excludedIndices.size() > 1` check. The setting should not be set to blank/empty string in any case. This will cost an extra call to Regex.simpleMatch() for every request. To clear the setting, users can set the value to `[]` or `null` instead of "".
```
public void validateExcludedIndices(@NonNull List<String> excludedIndices) {
    for (String index : excludedIndices) {
        if (index == null) {
            throw new IllegalArgumentException("Excluded index name cannot be null.");
        }
        if (index.isBlank()) {
            throw new IllegalArgumentException("Excluded index name cannot be blank.");
        }
    }
}
```